server/src/modules/jobs/validation/hydrate_from_constructys.ts

```diff
@@ -0,0 +1,68 @@
+import { PassThrough } from "node:stream";
+
+import { addJob } from "job-processor";
+import { ObjectId } from "mongodb";
+import { DOCUMENT_TYPES } from "shared/constants/documents";
+
+import parentLogger from "@/common/logger";
+
+import { initFtpConnection } from "../../../common/utils/ftpUtils";
+import { DEFAULT_DELIMITER } from "../../../common/utils/parserUtils";
+import config from "../../../config";
+import { createUploadDocument, updateDocument, uploadFile } from "../../actions/documents.actions";
+const logger = parentLogger.child({ module: "job:validation:hydrate_from_constructys" });
+
+export const run_hydrate_from_constructys = async () => {
+  logger.info("Constructys data import starting...");
+
+  logger.info("Get remote file...");
+
+  const document = await createUploadDocument({
+    type_document: DOCUMENT_TYPES.CONSTRUCTYS,
+    fileSize: 0,
+    filename: "constructys-data.csv",
+    delimiter: DEFAULT_DELIMITER,
+    added_by: new ObjectId(),
+  });
+
+  const myTransform = new PassThrough({
+    construct(callback) {
+      // @ts-expect-error
+      this.fileSize = 0;
+      callback();
+    },
+    transform(chunk, _encoding, callback) {
+      // @ts-expect-error
+      this.fileSize += chunk.length;
+      callback(null, chunk);
+    },
+  });
```

Comment:
```suggestion
 let fileSize = 0;
  const myTransform = new PassThrough({
    transform(chunk, _encoding, callback) {
     fileSize += chunk.length;
      callback(null, chunk);
    },
  });
```

Tu connais pas les champs possible et l'evolution, le mieux c'est de pas touché à la classe